src/main/java/com/google/sps/data/Chat.java

```diff
@@ -42,54 +42,123 @@
   // //Constructor 2
   Chat(Entity entity) {
     this.id = (long) entity.getKey().getId();
-    this.participants = entity.getString("participants");
-    // this.messages = entity.getProperty("messages");
+    this.participants = entity.getList("participants");
+    this.messages = entity.getList("messages");
+
   }
 
-  public static Chat getChat(Datastore datastore, String chatid) {
-    Key key = datastore.newKeyFactory().setKind("Chat").newKey(chatid);
+    public static Chat newChat(Datastore datastore, String participantId1, String participantId2) {
+        boolean chat_exists = Chat.chatExists(datastore, participantId1, participantId2);
```

Comment:
[Local variable names are written in lowerCamelCase.](https://google.github.io/styleguide/javaguide.html#s5.2.7-local-variable-names)